server/base/src/main/java/org/apache/accumulo/server/metadata/iterators/TabletMetadataCheckIterator.java

```diff
@@ -98,11 +97,11 @@ public void seek(Range range, Collection<ByteSequence> columnFamilies, boolean i
 
     var colsToRead = check.columnsToRead();
 
-    source.seek(new Range(tabletRow), Set.of(), false);
+    source.seek(new Range(tabletRow), colsToRead.getFamilies(), true);
 
     if (source.hasTop()) {
       var tabletMetadata = TabletMetadata.convertRow(new IteratorAdapter(source),
-          EnumSet.copyOf(colsToRead), false, false);
+          EnumSet.copyOf(colsToRead.getColumns()), false, false);
```

Comment:
Could move this copy of the set into ResolvedColumns.getColumns() for encapsulation.  Then the real internal set never escapes.